### ISSUE TITLE
Update pytest to 6.2.5

### ISSUE THREAD

**Issue by rtibbles** [learningequality/kolibri#11728]
Our version of pytest was pinned to a very old version for back compatibility with Python 2.7.

Now that we have dropped support for 2.7, we can upgrade pytest to versions that support Python 3.6 and above.

While technically, 7.0.1 may work, there have been issues reported with Python 3.10 (cc @jredrejo), so it seems safer to upgrade to 6.2.5, as we would not be able to upgrade to newer versions of the 7 series, as they drop support for Python 3.6, which we still support.

We would also need to update any pytest plugins, and remove the existing [patching of pytest for Python 3.10 compatibility](https://github.com/learningequality/kolibri/blob/release-v0.16.x/test/patch_pytest.py).

```[tasklist]
### Acceptance Criteria
- [x] Update pytest and pytest plugins
- [x] Remove patching for pytest for Python 3.10 compatibility
```

**Comment by thesujai** [comment]
I would like to give this a try.
Can i be assigned to this one?

**Comment by AlexVelezLl** [comment]
Hi @thesujai! Sure! I have assigned this issue to you :open_hands:. Let us know if you have any questions.

**Comment by thesujai** [comment]
@rtibbles how do i approach this one. How I amd doing is going through https://docs.pytest.org/en/stable/changelog.html and checking for the Removals.

**Comment by rtibbles** [comment]
There are two ways of doing it - you can go through the changelog and check for removals, or you can just do the upgrade and then run pytest locally to see what breaks!

I would suggest a hybrid strategy - get a good sense of what has changed in the Changelog, and if you can see any obvious things that you can preempt that might be a problem with the upgrade, try to fix those alongside the initial upgrade. Then try out running pytest and see what breaks to start to fix it.

To reduce your iteration time, you can also run pytest with only a single test file (it shouldn't matter too much which one it is, but maybe try a couple).

**Comment by rtibbles** [comment]
Fixed in #11759
